src/routes/v2/sgidAuth.ts

```diff
@@ -86,25 +90,130 @@ export class SgidAuthRouter {
 
     const { nonce, codeVerifier } = cookieData
 
-    // Exchange the authorization code and code verifier for the access token, then use the access token to retrieve user's email
-    await this.sgidAuthService
+    // Exchange the authorization code and code verifier for the access token, then use the access token to retrieve user's data
+    const userDataRes = await this.sgidAuthService
       .retrieveSgidAccessToken({ authCode, nonce, codeVerifier })
       .andThen(({ accessToken, sub }) =>
         // We can immediately process the access token - we only need to retrieve the email from sgid
-        this.sgidAuthService.retrieveSgidUserEmail(accessToken, sub)
+        this.sgidAuthService.retrieveSgidUserEmails(accessToken, sub)
       )
-      .andThen((email) => {
-        if (!email || !email.endsWith("@open.gov.sg")) {
+      .andThen((officerDetails) => {
+        if (!officerDetails || officerDetails.length === 0) {
           return errAsync(
             new AuthError(
               "Your email has not been whitelisted to use sgID login. Please use another method of login."
             )
           )
         }
-        return okAsync(email)
+        return okAsync(officerDetails)
       })
-      .andThen((email) =>
-        // Login with email
+    if (userDataRes.isErr()) {
+      const { error } = userDataRes
+      if (
```

Comment:
No error messages for this whole chunk - could we put in something meaningful to the user? 

Otherwise, they'd be redirected with no toast on why

Reply:
The reason why no error messages are sent is because these endpoints only show up on the temporary redirect-in-progress page, which users should never see a toast/error message on. Our current behaviour is that the frontend handles the error messages according to status code (we only get 500/401 errors) upon the user getting redirected to the login page.

Reply:
hmm ok - i was hoping for the frontend to extract the error message out from teh backend isntead of it deriving the message from status code. 

is this a fast change? if not, fine iwth leaving as is and doing lkater

Reply:
Let's do this in a separate PR?